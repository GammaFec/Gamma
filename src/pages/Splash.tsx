import React from "react";
import Logo from "../img/Logo.jpg";
import { StyledContainer, StyledImg, StyledParagraph } from "./Styles";

const Splash = () => {
    return (
        <StyledContainer>
            <div>
                <StyledImg alt="Logo de adoptemos" src={Logo} />
                <StyledParagraph>Adogtemos</StyledParagraph>
            </div>
        </StyledContainer>
    );
};

export default Splash;
