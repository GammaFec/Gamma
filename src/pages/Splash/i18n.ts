import i18next from "i18next";

import { LanguageList } from "./../../i18next/config";

i18next.addResources(LanguageList.SPANISH, "Splash", {
    ImageAlt: "Logo de Adogtemos"
});

i18next.addResources(LanguageList.ENGLISH, "Splash", {
    ImageAlt: "Adogtemos Logo"
});
