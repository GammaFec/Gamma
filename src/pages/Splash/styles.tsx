import styled from "styled-components";
import { fontSizes, theme } from "../../common/styles/variables";

export const StyledContainer = styled.div`
    width: 100%;
    min-height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const StyledImg = styled.img`
    width: 200px;
    height: 140px;
    display: block;
    margin: auto;
`;

export const StyledParagraph = styled.p`
    font-family: ${theme.fontPrimary};
    text-align: center;
    margin-top: 30px;
    font-size: ${fontSizes.font24};
    font-weight: 900;
`;
