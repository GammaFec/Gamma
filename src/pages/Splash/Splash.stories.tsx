/* eslint-disable @typescript-eslint/no-empty-function */
import React from "react";
import { Story, Meta, addDecorator } from "@storybook/react";
import StoryRouter from "storybook-react-router";

import Splash from ".";

export default {
    title: "Pages/Splash",
    component: Splash,
    parameters: {
        docs: {
            description: {
                component: "Splash page"
            }
        }
    }
} as Meta;

addDecorator(StoryRouter());

const Template: Story = () => <Splash />;

export const Primary = Template.bind({});
