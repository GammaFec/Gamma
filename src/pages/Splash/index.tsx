import React from "react";
import Logo from "../../img/Logo.svg";
import { StyledContainer, StyledImg, StyledParagraph } from "./styles";

const Splash: React.FunctionComponent = () => {
    return (
        <StyledContainer>
            <div>
                <StyledImg alt="Logo de adoptemos" src={Logo} />
                <StyledParagraph>Adogtemos</StyledParagraph>
            </div>
        </StyledContainer>
    );
};

export default Splash;
