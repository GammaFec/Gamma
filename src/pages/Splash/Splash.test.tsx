import React from "react";
import Splash from "./";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

jest.mock("react-i18next", () => ({
    useTranslation: (): { t: (key: string) => string } => ({ t: (key: string): string => key })
}));

describe("<Splash />", () => {
    test("Render", () => {
        const { asFragment } = render(<Splash />);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Render all childrens", () => {
        const { getByText, getByAltText } = render(<Splash />);

        getByAltText("ImageAlt");
        getByText("Adogtemos");
    });
});
