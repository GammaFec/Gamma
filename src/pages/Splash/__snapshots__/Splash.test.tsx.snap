// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> Render 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 200px;
  height: 140px;
  display: block;
  margin: auto;
}

.c2 {
  font-family: 'Poppins','sans-serif';
  text-align: center;
  margin-top: 30px;
  font-size: 24px;
  font-weight: 900;
}

<div
    class="c0"
  >
    <div>
      <img
        alt="ImageAlt"
        class="c1"
        src="Logo.svg"
      />
      <p
        class="c2"
      >
        Adogtemos
      </p>
    </div>
  </div>
</DocumentFragment>
`;
